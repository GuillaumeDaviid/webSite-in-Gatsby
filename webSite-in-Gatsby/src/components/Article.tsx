import * as React from 'react'
import {useState} from 'react'
import './styles/Article.scss'
import { Link, graphql, StaticQuery } from 'gatsby'
import { GatsbyImage } from 'gatsby-plugin-image'


export default function Article({data}:any) {
  const [p, setp] = useState(0);
  const [nbr, setNbr] = useState(1);

  console.log(data)

  function handleClickNext() {
    let carrousel = document.getElementById('carrousel');
    if (carrousel){
        if (nbr < 8){
            setNbr(nbr + 1);
            const newp = p - 1
            setp(newp);
            carrousel.style.transform = "translate("+newp*25+"vw)";
            carrousel.style.transition = "all 0.5s ease";
        }
    }
}

function handleClickPrevious() {
    let carrousel = document.getElementById('carrousel');
    if (carrousel){
        if (nbr > 1){
            setNbr(nbr - 1);
            const newp = p + 1;
            setp(newp);
        carrousel.style.transform = "translate("+newp*25+"vw)";
        carrousel.style.transition = "all 0.5s ease";
        }
        
    }
}


    return(
        <div className="Article">
         
            <h2 data-testid="title">Article</h2>
            <div className="Article_separator" data-testid="separator"></div>
            
            <div className="Article_items">
                <button className="Article_items-btn right" onClick={handleClickPrevious} data-testid="btn">&#60;</button>
                <button className="Article_items-btn left" onClick={handleClickNext} data-testid="btn">&#62;</button>

                <div className='Article_items-elt' id="carrousel">


<StaticQuery
      query={graphql`
      query {
        allMdx(sort: {fields: frontmatter___date, order: DESC}, limit: 10) {
          nodes {
            frontmatter {
              date(formatString: "MMMM D, YYYY")
              title
              slug
              featuredImage {
                childImageSharp {
                  gatsbyImageData(height: 100)
                }
              }
            }
            id
            excerpt
          }
        }
      }
      `}
      render={data => (
        <div >
          <div className='Article_items-elt'>{data.allMdx.nodes.map((node:any) => (
              <div className="Article_items-elt-item">
            <GatsbyImage image={node.frontmatter.featuredImage.childImageSharp.gatsbyImageData} alt='asdsf' className='blog_image'/>
            
            <h2 className='item_link'>
              <Link to={`/blog/${node.frontmatter.slug}`}>
                {node.frontmatter.title}
              </Link>
            </h2>
           
         </div>
        ))}<div className='Article_more'><Link to={'/blog'}>Voir plus d'articles</Link></div>
        </div>
        </div>
      )}
    />

                </div>
            </div>
        </div>
    )
}