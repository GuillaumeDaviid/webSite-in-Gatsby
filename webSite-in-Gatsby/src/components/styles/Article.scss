.Article {
    text-align: center;

    &_separator {
        height: 2px;
        width: 85px;
        margin: 70px auto;
        background-color: #222222;
    }

    &_items {
        min-height: 300px;
        overflow: hidden;
        margin-bottom: -20px;

        &-btn {
            z-index: 99;
            position: absolute;
            cursor: pointer;
            height: 300px;
            color: black;
            border-color: grey;
            background: rgba(255, 255, 255, .5);
            width: 30px;
        }

        &-elt {
            position: relative;
            display: flex;
            
            &-item {
                min-width: 25vw;
            }
        }
    }

    &_more {
        font-weight: 900;
        display: flex;
        justify-content: center;
        align-items: center;
        min-width: 25vw;
        color: #337ab7;
    }
}

.right {
    left: 0vw;
}

.left {
    right: 0vw;
}

.item {
    &_img {
        max-width: 200px !important;
        max-height: 100px !important;
    }
    &_link {
        font-size: 2rem;
        color: #222222;
    }
}


@media (max-width: 750px) {
    .Article {
      &_items {
        &-elt {
            display: flex;
            flex-direction: column;
        }

        &-btn {
            display: none;
        }
      }

      &_more {
        margin-bottom: 20px;
      }
    }
}

@media (max-width: 400px) {
    .item {
        &_link {
            font-size: 1.5rem;
        }
    }
}