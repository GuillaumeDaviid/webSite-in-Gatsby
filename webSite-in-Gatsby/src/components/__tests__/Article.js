import React from "react"
import renderer from "react-test-renderer"
import Article from "../Article.tsx"
import { render, screen } from "@testing-library/react"

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Article />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should have a title element", () => {
    render(<Article />)

    const title = screen.getByTestId('title');
    expect(title).toBeTruthy();
  })

  it("should have a separator element", () => {
    render(<Article />)

    const separator = screen.getByTestId('separator');
    expect(separator).toBeTruthy();
  })

  it("should have 2 buttons to switch", () => {
    render(<Article />)

    const btn = screen.getAllByTestId('btn');
    expect(btn).toHaveLength(2);
  })

  // test handleclick et data avec graphQL
})