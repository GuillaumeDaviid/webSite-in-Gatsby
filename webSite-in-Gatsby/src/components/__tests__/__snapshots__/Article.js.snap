// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="Article"
>
  <h2
    data-testid="title"
  >
    Article
  </h2>
  <div
    className="Article_separator"
    data-testid="separator"
  />
  <div
    className="Article_items"
  >
    <button
      className="Article_items-btn right"
      data-testid="btn"
      onClick={[Function]}
    >
      &lt;
    </button>
    <button
      className="Article_items-btn left"
      data-testid="btn"
      onClick={[Function]}
    >
      &gt;
    </button>
    <div
      className="Article_items-elt"
      id="carrousel"
    />
  </div>
</div>
`;
